scripts/check_repo_stats.py

```diff
@@ -0,0 +1,174 @@
+"""
+This script checks if a repo fulfills certain requirements
+Can customize what requirements to check and values to check against
+
+
+References
+
+Parts 1 and 2 of a medium blog explaining the query:
+https://fabiomolinar.medium.com/using-githubs-graphql-to-retrieve-a-list-of-repositories-their-commits-and-some-other-stuff-ccbbb4e96d78
+https://fabiomolinar.medium.com/using-githubs-graphql-to-retrieve-a-list-of-repositories-their-commits-and-some-other-stuff-ce2f73432f7
+
+Repository Object:
+https://docs.github.com/en/graphql/reference/objects#repository
+
+"""
+from datetime import datetime
+import fire
+import sys
+
+from common import get_graphql_data
+
+
+#### Requirement Callables ####
+def req_enough_stars(metadata: dict, req_stars: str = "10") -> bool:
+    """Checks if Github repository has enough stars"""
+    return metadata["stargazerCount"] >= int(req_stars)
+
+
+def req_latest_commit(metadata: dict, date_str: str = "2020-1-1") -> bool:
+    """Checks if Github repository has a valid latest commit"""
+    # latest_commit_date = datetime.fromisoformat(metadata["pushedAt"])
+    date_values = metadata["pushedAt"].split("T")[0].split("-")
+    latest_commit_date = datetime(
+        int(date_values[0]), int(date_values[1]), int(date_values[2])
+    )
+    date = date_str.split("-")
+    req_date = datetime(int(date[0]), int(date[1]), int(date[2]))
+    return latest_commit_date.date() > req_date.date()
+
+
+def req_language(metadata: dict, language: str = "java") -> bool:
+    """Checks if Github repository has correct language"""
+    return metadata["primaryLanguage"]["name"].lower() == language.lower()
+
+
+def req_fuzzers(metadata: dict) -> bool:
+    """ "Checks if Github repository of Rust has fuzz file"""
+    contents = metadata["object"]["entries"]
+    for item in contents:
+        if item["name"] == "fuzz" and item["type"] == "tree":
+            return True
+    return False
+
+
+#### End Requirement Callables ####
+
+
+def check_requirements(
+    repo: str, requirements: list[callable], reqs: list[str], metadata: dict = None
```

Comment:
let `requirements` have type `Callable[[dict, ...], bool]` to tell it requires at least a metadata dict and must return a bool